handlers/serviceset.go

```diff
@@ -167,6 +176,22 @@ func (s *ServiceSet) logError(format string, args ...interface{}) {
 	}
 }
 
+// reloadEndpointHandler
+func (s *ServiceSet) reloadEndpointHandler(w http.ResponseWriter, r *http.Request) {
+	token, err := r.URL.Query().Get("token")
+	if s.enableReloadEndpoint && token == s.reloadToken {
```

Comment:
I'd say that not finding the token should return a 400 ("deceptive request routing") while the endpoint not being enabled should return a 403 (forbidden). 404 doesn't look bad either for the disabled endpoint case, but I like having different errors. This way you can easily tell whether this is a server-side configuration problem (server not run with the right option), client-side configuration (wrong token) or wrong path (404, the default).